jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/IO.java

```diff
@@ -224,36 +224,110 @@ public static void copyDir(File from, File to) throws IOException
     }
 
     /**
-     * Copy the contents of a directory from one directory to another.
+     * Copy the contents of a source directory to destination directory.
      *
      * @param srcDir the source directory
      * @param destDir the destination directory
      * @throws IOException if unable to copy the file
      */
+    public static void copyDir(Path srcDir, Path destDir) throws IOException
+    {
+        if (!Files.isDirectory(Objects.requireNonNull(srcDir)))
+            throw new IllegalArgumentException("Source is not a directory: " + srcDir);
+        if (!Files.isDirectory(Objects.requireNonNull(destDir)))
+            throw new IllegalArgumentException("Dest is not a directory: " + destDir);
+
+        try (Stream<Path> sourceStream = Files.walk(srcDir))
+        {
+            Iterator<Path> iterFiles = sourceStream
+                .filter(Files::isRegularFile)
+                .iterator();
+            while (iterFiles.hasNext())
+            {
+                Path sourceFile = iterFiles.next();
+                Path relative = srcDir.relativize(sourceFile);
+                Path destFile = resolvePath(destDir, relative);
+                if (!Files.exists(destFile.getParent()))
+                    Files.createDirectories(destFile.getParent());
+                copyFile(sourceFile, destFile);
+            }
+        }
+    }
+
+    /**
+     * Copy the contents of a source directory to destination directory.
+     *
+     * @param srcDir the source directory
+     * @param destDir the destination directory
+     * @param copyOptions the options to use on the {@link Files#copy(Path, Path, CopyOption...)} commands.
+     * @throws IOException if unable to copy the file
+     */
     public static void copyDir(Path srcDir, Path destDir, CopyOption... copyOptions) throws IOException
     {
         if (!Files.isDirectory(Objects.requireNonNull(srcDir)))
             throw new IllegalArgumentException("Source is not a directory: " + srcDir);
         if (!Files.isDirectory(Objects.requireNonNull(destDir)))
             throw new IllegalArgumentException("Dest is not a directory: " + destDir);
 
-        try (Stream<Path> sourceStream = Files.walk(srcDir, 20))
+        try (Stream<Path> sourceStream = Files.walk(srcDir))
         {
             Iterator<Path> iterFiles = sourceStream
                 .filter(Files::isRegularFile)
                 .iterator();
             while (iterFiles.hasNext())
             {
                 Path sourceFile = iterFiles.next();
-                URI relativeSrc = srcDir.toUri().relativize(sourceFile.toUri());
-                Path destFile = destDir.resolve(relativeSrc.toASCIIString());
+                Path relative = srcDir.relativize(sourceFile);
```

Comment:
Removing the conversion from `Path` to `URI` then back to `Path` has some merit, but I wonder why we did that in the first place and I'm worried this change may break something in a subtle way.

We need to understand the why we were converting to URI, if getting rid of that gives identical results and if not, write a test that asserts the difference.

Reply:
If we use `Path.resolve(Path)` and the two `Path` references are on different FileSystem's then the `Path.resolve(Path)` fails with an Exception.
The URI trick was here so that `Path.resolve(String)` could be used instead.
Turns out, there is a completely different technique that could be used to avoid the Exception.
See the old `Resource.resolveDifferentFileSystem(Path, Path)` technique, now `IO.resolvePath()`

Reply:
Got it, thanks.